core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -23,7 +23,6 @@ import java.text.DateFormat
 import java.util.{Arrays, Date, Locale}
 
 import scala.collection.JavaConverters._
-import scala.collection.immutable.Map
```

Comment:
I'm not sure we can remove these - I forget, but they are probably necessary in Scala 2.13

Reply:
Let me check it manually

Reply:
I check Scala 2.13 as follows:

```
dev/change-scala-version.sh 2.13
mvn clean install -pl core -Pscala-2.13 -am -Dtest.exclude.tags=org.apache.spark.tags.ExtendedLevelDBTest
```



```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Spark Project Parent POM 3.4.0-SNAPSHOT:
[INFO] 
[INFO] Spark Project Parent POM ........................... SUCCESS [  4.233 s]
[INFO] Spark Project Tags ................................. SUCCESS [  8.936 s]
[INFO] Spark Project Local DB ............................. SUCCESS [ 12.578 s]
[INFO] Spark Project Networking ........................... SUCCESS [ 53.431 s]
[INFO] Spark Project Shuffle Streaming Service ............ SUCCESS [ 12.492 s]
[INFO] Spark Project Unsafe ............................... SUCCESS [ 10.007 s]
[INFO] Spark Project Launcher ............................. SUCCESS [ 10.775 s]
[INFO] Spark Project Core ................................. SUCCESS [33:39 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  35:32 min
[INFO] Finished at: 2022-07-30T22:29:15+08:00
[INFO] ------------------------------------------------------------------------
```

Build and test passed and I also check in branch 2.13 that Predef still exists

https://github.com/scala/scala/blob/cbc012e573346dc685c478eec5fcbb56d22ea884/src/library/scala/Predef.scala#L157-L164

Reply:
Hm, I think the problem is that it compiles both ways, but means different things. I do think we want the explicit import of Map for Scala 2.13 so that it does not mean scala.collection.Map in either build.

Reply:
I'm not sure if I don't understand what you mean,  but `Map[K, +V]` represents `immutable.Map[K, V]` in both Scala 2.12 and Scala 2.13 (From the [migration document](https://docs.scala-lang.org/overviews/core/collections-migration-213.html, only `scala.Seq[+A]` changed from `scala.collection.Seq[A]` to `scala.collection.immutable.Seq[A]`, `scala.Map[K, +V]`  and `scala.Set[A]` didn't.)

In addition, where `scala.collection.Map` needs to be used in spark code, we now explicitly imports it.

Reply:
However, I think it is also good to explicitly imports `immutable.Map[K, V]` and `immutable.Set[A]`,  which makes it more clearer

Reply:
Let us just leave the explicitly imports, I will close this pr

Reply:
You're probably right, that it's just unused. I'm neutral on it, just because it's just an unused import